snapcraft/tests/commands/test_help.py

```diff
@@ -69,6 +69,15 @@ def test_print_module_help_for_valid_plugin(self):
                          'The help message does not start with {!r} but with '
                          '{!r} instead'.format(expected, output))
 
+    def test_print_module_named_with_dashes_help_for_valid_plugin(self):
+        main(['help', 'plainbox-provider'])
+
+        expected = ' Create parts'
+        output = self.fake_terminal.getvalue()
+        self.assertThat(output, tests.testtools.matchers.StartsWith(expected),
+                         'The help message does not start with {!r} but with '
+                         '{!r} instead'.format(expected, output[:len(expected)]))
```

Comment:
Great! 
Just one more minor detail. testtools takes care of the formatting of the error message, so you can remove there the third argument. Give it a try forcing the test to fail with the wrong expected text.